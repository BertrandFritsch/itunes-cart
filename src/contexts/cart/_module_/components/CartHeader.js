// @flow

import React from 'react';

type Props = {
  /**
   * The number of items in the cart
   */
    itemCount: number,

  /**
   * The total price
   */
    totalPrice: number,

  /**
   * Notify the user wants checkout
   */
    onCheckout: () => void
};

/**
 * CartHeader Component
 *
 * Use a pure function
 */
export default function CartHeader(props: Props) {
  return props.itemCount > 0
      ? <div className="cart-item">
          <span className="cart-header-item-count">{`${props.itemCount} item${props.itemCount > 1 ? 's' : ''} - `}</span>
          <span className="cart-header-total-price">{`${Math.round(props.totalPrice * 100) / 100} $`}</span>
          <button className="cart-header-checkout" onClick={ () => props.onCheckout() }>Checkout</button>
        </div>
      : <div className="cart-header">
          <span className="cart-header-empty-place-holder">The cart is empty</span>
        </div>;
}
