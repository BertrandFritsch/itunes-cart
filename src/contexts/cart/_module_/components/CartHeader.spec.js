import React from 'react';
import renderer from 'react-test-renderer';
import CartHeader from './CartHeader';

describe('CartHeader', () => {
  it('should render correctly', () => {
    const props = {
      itemCount: 0,
      totalPrice: 0,
      onCheckout: () => {}
    };

    const component = renderer.create(
      <CartHeader { ...props } />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render correctly also with a non-empty cart', () => {
    const props = {
      itemCount: 2,
      totalPrice: 1.98,
      onCheckout: () => {}
    };

    const component = renderer.create(
      <CartHeader { ...props } />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should dispatch the checkout event', () => {
    const props = {
      itemCount: 2,
      totalPrice: 1.98,
      onCheckout: jest.fn()
    };

    const component = mount(
      <CartHeader { ...props } />
    );

    component.find('button').simulate('click', { target: { } });
    expect(props.onCheckout).toBeCalledWith();

  });
});
