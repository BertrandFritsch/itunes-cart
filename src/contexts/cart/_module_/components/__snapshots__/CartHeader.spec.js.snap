// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartHeader should render correctly 1`] = `
<div
  className="cart-header"
>
  <span
    className="cart-header-empty-place-holder"
  >
    The cart is empty
  </span>
</div>
`;

exports[`CartHeader should render correctly also with a non-empty cart 1`] = `
<div
  className="cart-item"
>
  <span
    className="cart-header-item-count"
  >
    2 items - 
  </span>
  <span
    className="cart-header-total-price"
  >
    1.98 $
  </span>
  <button
    className="cart-header-checkout"
    onClick={[Function]}
  >
    Checkout
  </button>
</div>
`;
